lib/rex/post/meterpreter/extensions/stdapi/fs/dir.rb

```diff
@@ -148,18 +148,24 @@ def Dir.match(name, dir = false)
     end
 
     fpath.each_with_index do |file_name, idx|
-      if dir && sbuf[idx]
+      is_dir = false
+      if sbuf[idx]
         st = ::Rex::Post::FileStat.new
         if new_stat_buf
           st.update(sbuf[idx].value)
         else
           st.update32(sbuf[idx].value)
         end
-        next if st.ftype != 'directory' # if file_name isn't directory
+        is_dir = st.ftype == 'directory'
+        next if (dir && !is_dir) # if file_name isn't directory
```

Comment:
Just a comment here but would be nice to rename the `dir` variable to something like `match_dir` to better convey its purpose now that we have an `is_dir` variable here. Otherwise as it is right now its very confusing to determine what `dir` is meant to be used for without reading the comments at the top of the function definition.

Reply:
I can fix this up before we land this though, so no issue.

Reply:
Fixed in e6f38b3226f0e2375cf6c371d42d8452974ec22c